secrets-provider-test-chart/templates/serviceaccount.yaml

```diff
@@ -0,0 +1,6 @@
+{{- if eq .Values.rbac.serviceAccount.create true -}}
```

Comment:
I think we shouldn't create and bind the role if we don't create the service account.
So this ccheck should be in both `role` and `rolebinding`.